R/lazyframe-frame.R

```diff
@@ -1006,11 +1006,10 @@ lazyframe__fill_null <- function(
     }
     check_dots_empty0(...)
     if (!missing(value) && !is.null(value)) {
+      dtype <- infer_polars_dtype(value)
```

Comment:
I think this would be an error if we passed a polars_expr for `value`. Test cases are needed.